### PR TITLE
Let iframes accept a LocalClip

### PR DESCRIPTION
**Pull request by mrobinson**
This will make it easier for Servo iframes to clip their contents. The
LocalClip is converted into a containing ClipScrollTree node around the
content of the iframe.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1472)

### CODE REVIEW

File: webrender/src/frame.rs [hunk 1]
```diff
@@ -495,6 +496,17 @@ impl Frame {
             None => return,
         };
 
+        let mut clip_region = ClipRegion::for_clip_node_with_local_clip(local_clip);
+        clip_region.origin += reference_frame_relative_offset;
+        let parent_pipeline_id = parent_id.pipeline_id();
+        let clip_id = self.clip_scroll_tree.generate_new_clip_id(parent_pipeline_id);
+        println!("clip: {:?}", clip_region);
```
Comment (kvark) [inline webrender/src/frame.rs]: leftover `println`

File: webrender/src/mask_cache.rs [hunk 2]
```diff
@@ -44,6 +44,14 @@ impl ClipRegion {
         }
     }
 
+    pub fn for_clip_node_with_local_clip(local_clip: &LocalClip) -> ClipRegion {
```
Comment (glennw) [inline webrender/src/mask_cache.rs]: The naming of this function seems a little unclear to me.

Comment (mrobinson) [inline webrender/src/mask_cache.rs]: Okay. I changed this to create_for_clip_node_with_local_clip. I'll open a PR to make the other names coherent with it as well.

**Pull request merged**